cmd/otel-allocator/allocation/http.go

```diff
@@ -30,20 +30,13 @@ func GetAllTargetsByJob(job string, cMap map[string][]TargetItem, allocator *All
 
 			var targetGroupList []targetGroupJSON
 
-			trg := make(map[string][]TargetItem)
 			for _, t := range targetList {
-				trg[t.JobName+t.Label.String()] = append(trg[t.JobName+t.Label.String()], t)
+				targetGroupList = append(targetGroupList, targetGroupJSON{
+					Targets: []string{t.TargetURL},
+					Labels:  t.Label,
```

Comment:
Does this change need to be applied to `GetAllTargetsByCollectorAndJob` as well?  Do tests need to be added/modified to ensure that they are validating the correct data are produced?

Reply:
Yes, `GetAllTargetsByCollectorAndJob` also needs to be changed, I will add the processing of `GetAllTargetsByCollectorAndJob` later, and add related tests

Reply:
> Does this change need to be applied to `GetAllTargetsByCollectorAndJob` as well? Do tests need to be added/modified to ensure that they are validating the correct data are produced?

Thanks for the tip, the change is done